src/utils.ts

```diff
@@ -4,6 +4,10 @@ export const toStars = (addr: string) => {
   if (!addr.startsWith('stars')) {
     const { data } = fromBech32(addr);
     const starsAddr = toBech32('stars', data);
+    // wallet address length 20, contract address length 32
+    if (![20, 32].includes(data.length)) {
+      throw new Error('Invalid address: ' + addr + ' ' + starsAddr);
+    }
```

Comment:
Since you use this function for validation now, you should do all the validation outside of the `if`. Otherwise you can bypass the validation wioth an input like `star1muhahah I'm broken and you don't detect me`.

Reply:
good point, validating all addresses